### PR TITLE
Species-Empire Opinions and Annexation

### PR DESCRIPTION
**Pull request by geoffthemedio**
-add species opinions of empires and effects that modify them
-add annexation as a way to take control of planets without invasion or colonization

### CODE REVIEW

**Comment by o01eg** [comment]
I suppose `COMMON_OPINION_EFFECTS` macros should be implemented for Python FOCS too?

**Comment by geoffthemedio** [comment]
> I suppose `COMMON_OPINION_EFFECTS` macros should be implemented for Python FOCS too?

I don't know if / what is needed specifically for Python FOCS stuff.

**Comment by o01eg** [comment]
I put it into species implemented with Python FOCS but unsure about values.

**Comment by geoffthemedio** [comment]
I think I've added enough stuff now to this to consider testing an possibly merging into master.

Updating the Python parsers with all the stuff that's been added to the C++ one would be good. That would eg. allow different species to have different costs and conditions for annexation.

**Comment by o01eg** [comment]
I'll fix Python FOCS parser and content soon, hopefully this week.

**Review by o01eg — APPROVED** [review]
Made final touch in Python FOCS part.

File: parse/ConditionPythonParser.cpp [hunk 1]
```diff
@@ -788,6 +788,8 @@ void RegisterGlobalsConditions(boost::python::dict& globals) {
     globals["IsSource"] = condition_wrapper(std::make_shared<Condition::Source>());
     globals["IsTarget"] = condition_wrapper(std::make_shared<Condition::Target>());
     globals["IsRootCandidate"] = condition_wrapper(std::make_shared<Condition::RootCandidate>());
+    globals["IsAnyObject"] = condition_wrapper(std::make_shared<Condition::All>());
+    globals["NoObject"] = condition_wrapper(std::make_shared<Condition::None>());
```
Comment (geoffthemedio) [inline parse/ConditionPythonParser.cpp:792]: @o01eg OK? Is there a better alternative name for this? Obviously `None` won't work in Python...

Comment (o01eg) [reply parse/ConditionPythonParser.cpp:792]: Yes, `NoEffect` is for the same.

Comment (geoffthemedio) [reply parse/ConditionPythonParser.cpp:792]: note that `Condition::None` and `Condition::NoOp` are different

**Pull request merged**